src/WebJobs.Script.WebHost/Management/LinuxInstanceManager.cs

```diff
@@ -0,0 +1,153 @@
+                }
+
+                _logger.LogInformation($"Starting Assignment. Cloud Name: {_environment.GetCloudName()}");
+
+                // set a flag which will cause any incoming http requests to buffer
+                // until specialization is complete
+                // the host is guaranteed not to receive any requests until AFTER assign
+                // has been initiated, so setting this flag here is sufficient to ensure
+                // that any subsequent incoming requests while the assign is in progress
+                // will be delayed until complete
+                _webHostEnvironment.DelayRequests();
+
+                // start the specialization process in the background
+                Task.Run(async () => await AssignAsync(context));
+
+                return true;
+            }
+            else
+            {
+                // No lock needed here since _assignmentContext is not null when we are here
+                return _assignmentContext.Equals(context);
+            }
+        }
+
+        public abstract Task<string> ValidateContext(HostAssignmentContext assignmentContext);
+
+        private async Task AssignAsync(HostAssignmentContext assignmentContext)
+        {
+            try
+            {
+                // first make all environment and file system changes required for
+                // the host to be specialized
+                _logger.LogInformation("Applying {environmentCount} app setting(s)", assignmentContext.Environment.Count);
+                assignmentContext.ApplyAppSettings(_environment, _logger);
+                await ApplyContextAsync(assignmentContext);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Assign failed");
+                await _meshServiceClient.NotifyHealthEvent(ContainerHealthEventType.Fatal, GetType(), "Assign failed");
+                throw;
+            }
+            finally
+            {
+                // all assignment settings/files have been applied so we can flip
+                // the switch now on specialization
+                // even if there are failures applying context above, we want to
+                // leave placeholder mode
+                _logger.LogInformation("Triggering specialization");
+                _webHostEnvironment.FlagAsSpecializedAndReady();
+
+                _webHostEnvironment.ResumeRequests();
+            }
+        }
+
+        protected abstract Task ApplyContextAsync(HostAssignmentContext assignmentContext);
+
+        protected abstract Task<string> DownloadWarmupAsync(RunFromPackageContext context);
+
+        public IDictionary<string, string> GetInstanceInfo()
+        {
+            return new Dictionary<string, string>
+            {
+                { EnvironmentSettingNames.FunctionsExtensionVersion, ScriptHost.Version },
+                { EnvironmentSettingNames.WebsiteNodeDefaultVersion, "8.5.0" }
```

Comment:
Can we make this a const variable somewhere?

Reply:
Fixed! Thank you for your review!